src/Federation/EntityResolverProvider.php

```diff
@@ -252,13 +262,35 @@ protected function applySatisfiedSelection(EloquentBuilder $builder, SelectionSe
 
             $subSelection = $field->selectionSet;
             if (null === $subSelection) {
-                $builder->where($fieldName, $value);
+                $builder->where(
+                    $fieldName,
+                    $this->hasFieldWithDirective($definition, $fieldName, GlobalIdDirective::NAME)
+                        ? $this->globalId->decodeID($value)
+                        : $value
+                );
 
                 return;
             }
 
-            $this->applySatisfiedSelection($builder, $subSelection, $representation);
+            $this->applySatisfiedSelection($builder, $subSelection, $representation, $definition);
+        }
+    }
+
+    private function hasFieldWithDirective(ObjectTypeDefinitionNode $definition, string $fieldName, string $directiveName): bool
+    {
+        foreach ($definition->fields as $definitionField) {
+            if ($definitionField->name->value === $fieldName) {
+                foreach ($definitionField->directives as $directive) {
+                    if ($directive->name->value === $directiveName) {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
         }
+
+        return false;
```

Comment:
```suggestion
        $field = ASTHelper::firstByName($definition->fields, $fieldname);
        if (! $field) {
          return false;
        }

         return ASTHelper::hasDirective($field, $directiveName);
```

Reply:
Ah nice, couldn't find the helper, was searching for the wrong thing 😆 

Thanks!